### PR TITLE
Class refactor and method renames

### PR DESCRIPTION
**Pull request by apurvis**
replaces https://github.com/lumoslabs/broadside/pull/18

should be no functional changes and i have tested all the commands with the beacon staging deploy

@matl33t @rfroetscher

### CODE REVIEW

File: lib/broadside/configuration/deploy_config.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ class Configuration
     class DeployConfig < ConfigStruct
       include Utils
 
+      # TODO this shouldn't be the default; lots of apps using broadside are not rails
```
Comment (rfroetscher) [inline lib/broadside/configuration/deploy_config.rb]: this will break backwards compatability

**Comment by rfroetscher** [comment]
why the new PR?

File: lib/broadside/configuration/deploy_config.rb (hunk 1 above)
Comment (apurvis) [inline lib/broadside/configuration/deploy_config.rb]: yeah that's why it's a TODO... we'd have to go to 2.x to do this

**Comment by apurvis** [comment]
> why the new PR?

the old PR was a merge of this branch into my original feature branch, to keep the refactor and functionality separate.  unfortunately there's no way to change the destination of the merge to `master` without closing and reopening.

File: lib/broadside/configuration/deploy_config.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@ class Configuration
     class DeployConfig < ConfigStruct
       include Utils
 
-      # TODO this shouldn't be the default; lots of apps using broadside are not rails
+      # TODO for 2.x this shouldn't be the default; lots of apps using broadside are not rails
```
Comment (matl33t) [inline lib/broadside/configuration/deploy_config.rb]: will open an issue for this, so you can remove the comment

File: lib/broadside/deploy/ecs_deploy.rb [hunk 3]
```diff
@@ -150,141 +154,96 @@ def bash
...
+        w.max_attempts = nil
+        w.delay = config.ecs.poll_frequency
+        w.before_attempt do |attempt|
+          debug "Attempt #{attempt}: waiting for #{command_name} to complete..."
         end
+      end
+
+      info "#{command_name} task container logs:\n#{get_container_logs(task_arn)}"
+
+      # TODO: it should check ALL the task exit codes from all the runs
```
Comment (matl33t) [inline lib/broadside/deploy/ecs_deploy.rb]: shouldn't this be checking the exit code of only the application container?

Comment (apurvis) [inline lib/broadside/deploy/ecs_deploy.rb]: yeah i guess you are write; seems like it only actually runs on one container.  i'll delete this TODO

File: lib/broadside/configuration/deploy_config.rb [hunk 4]
```diff
@@ -57,24 +57,24 @@ def initialize
       # Checks existence of provided target
       def validate_targets!
         @targets.each do |target, configuration|
-          TARGET_ATTRIBUTE_VALIDATIONS.each do |var, validation|
+          invalid_messages = TARGET_ATTRIBUTE_VALIDATIONS.map do |var, validation|
```
Comment (matl33t) [inline lib/broadside/configuration/deploy_config.rb]: 🆒

**Comment by matl33t** [comment]
lgtm... merge whenever you're ready. may want to prune out some of those commit messages when you squash

**Comment by apurvis** [comment]
ran a manual test of all the commands with beacon; merged

**Pull request merged**